### ISSUE TITLE
Tracking Code

### ISSUE THREAD

**Issue by michaelwoodew** [Jenselme/aurelia-piwik#2]
how do you setup the _paq config for this plugin?
Or is this something that has to be implemented manually via the EventAggregator?

**Comment by Jenselme** [comment]
Like you would in any other application:
```js
var _paq = _paq || [];
    var siteId;
    var domains;
    if (/^www/.test(window.location.host)) {
        siteId = 3;
        domains = ["*.www.arenaoftitans.fr", "*.www.arenaoftitans.com"];
    } else {
        siteId = 4;
        domains = ["*.dev.arenaoftitans.com"]
    }
    (function () {
        var u = "//piwik.jujens.eu/";
        if (siteId === undefined) {
            return;
        }

        _paq.push(["setDomains", domains]);
        _paq.push(['setTrackerUrl', u + 'piwik.php']);
        _paq.push(['setSiteId', siteId]);
        _paq.push(['trackPageView']);
        _paq.push(['enableLinkTracking']);
    })();
```

**Comment by michaelwoodew** [comment]
I understand that, more specifically where would you place it with regards to Aureila?
Would you add this to app.js or app.html in a script tag for example.

**Comment by Jenselme** [comment]
I guess I didn't understand your question, sorry.

At first, I put in directly in the `index.html` in a script tag as described in the documenation of piwik. Now, I have [CSP](https://content-security-policy.com/) enabled so I put in into a `tracking.js` script that I load in the `index.html`.

I think it makes sense to put it in `main.js` or `app.js` to avoid loading an additional file. I wouldn't put it in `app.html` though, IMHO it makes more sense to put it in `index.html`.

To sum up, I don't really know. If you have an opinion on this, I'd be glad to hear you arguments.

**Comment by Jenselme** [comment]
Closing this. If you still have a question about this, please reopen.
